router/factory.go

```diff
@@ -5,8 +5,26 @@ import (
 	"github.com/rudderlabs/rudder-server/jobsdb"
 	"github.com/rudderlabs/rudder-server/services/rsources"
 	"github.com/rudderlabs/rudder-server/services/transientsource"
+	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
+type destinationConfig struct {
+	Name          string
+	ResponseRules map[string]interface{}
+	Config        map[string]interface{}
+	DestinationID string
+}
+
+func getRouterConfig(destination *backendconfig.DestinationT) destinationConfig {
+	var destConfig destinationConfig
+
+	destConfig.Name = destination.DestinationDefinition.Name
+	destConfig.DestinationID = misc.GetRouterIdentifier(destination.ID, destination.DestinationDefinition.Name)
+	destConfig.Config = destination.DestinationDefinition.Config
+	destConfig.ResponseRules = destination.DestinationDefinition.ResponseRules
+	return destConfig
+}
```

Comment:
We could live without this method, or the `destinationConfig` struct. All these can be extracted during `Setup` by passing the `backendconfig.DestinationT`

Reply:
Yeah true but passing the whole destinationT struct didn't feel right if we are going to use only a few parameters in it

Reply:
`backendconfig.DestinationT` is a config struct, so IMO it is relevant and it would help keep things simpler, but I don't insist. Can we convert `destinationConfig` fields to lowercase please?

Reply:
I saw this comment after adding my own about `RouterIdentifier`. I believe it is not the responsibility of backend-config to return the identifier used in our router logic. Yes, the destination as an abstraction is a good idea, but I would not put it inside `backendconfig.Destination`.